components/brave_wallet/browser/solana_utils.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/solana_utils.h"
+
+#include "base/check.h"
+#include "brave/third_party/bitcoin-core/src/src/base58.h"
+
+namespace brave_wallet {
+
+void CompactU16Encode(uint16_t u16, std::vector<uint8_t>* compact_u16) {
```

Comment:
As mentioned in the header, doc is at https://docs.solana.com/developing/programming-model/transactions#compact-u16-format.
Ref impl: https://github.com/solana-labs/solana-web3.js/blob/master/src/util/shortvec-encoding.ts#L15